src/Executor/DBHyphenationExecutor.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Executor;
+
+use App\Hyphenator\ArrayHyphenator;
+use App\Model\Rule;
+use App\Model\Word;
+use App\Processor\DBProcessor;
+use App\Repository\RuleRepository;
+use App\Repository\WordRepository;
+
+
+class DBHyphenationExecutor implements ExecutorInterface
+{
+    public function __construct(private readonly \PDO $connection, private readonly string $word)
+    {
+
+    }
+
+    public function execute(): void
+    {
+        $wordRepo = new WordRepository($this->connection);
+        $ruleRepo = new RuleRepository($this->connection);
+
+        $word = $wordRepo->findByWord($this->word);
+        if ($word !== null && $word->getHyphenated() !== null) {
+            $matchedRules = $ruleRepo->getRulesMatchingWord($word);
+            echo 'Hyphenated word: ' . $word->getHyphenated() . PHP_EOL;
+            foreach ($matchedRules as $rule) {
+                echo 'Matched rule: ' . $rule->getRule() . PHP_EOL;
+            }
+
+            return;
+        }
+
+        if ($word === null) {
+            $word = new Word($this->word);
+            $wordRepo->insertWord($word);
+            $word->setId((int) $wordRepo->getLastInsertedId());
+        }
+
+        $rules = array_map(function (Rule $rule) {
+            return $rule->getRule();
+        }, $ruleRepo->getRules());
```

Comment:
1. Syntax sugar could be applied (using FN)
2. Anonymous functions also can have a return type

```suggestion
        $rules = array_map(fn (Rule $rule): string => $rule->getRule(), $ruleRepository->getRules());
```

3. If using multiline then styling should be ajdusted (we use it in the project)

```php
        $rules = array_map(
            fn (Rule $rule): string => $rule->getRule(),
            $ruleRepository->getRules(),
        );
```